### PR TITLE
Hugo GitHub linking

### PR DESCRIPTION
**Pull request by SvenDowideit**
use the github linking patched hugo in `docker/hugo`

### CODE REVIEW

File: touch-up.sh [hunk 1]
```diff
@@ -33,74 +33,18 @@ do
...
-        -e 's/\(\][(]\)\(\.\.\/\)/\1\/'$y'\//g' {} \;         
       ;;
       *)
-         y=${i##*/}
-        find $i -type f -name "*.md" -exec sed -i.old \
-        -e '/^<!.*metadata]>/g' \
-        -e '/^<!.*end-metadata.*>/g' {} \;        
       ;;
       esac
 done
```
Comment (SvenDowideit) [inline touch-up.sh]: Yes, I'm whimping out - I haven't looked at what the next few lines do, but if we're lucky, we can replace them using Hugo meta data, or something from its `data/` dir support

Comment (thaJeztah) [inline touch-up.sh]: The `metadata/end-metadata` hack is for hiding the metadata on GitHub, iirc the distribution team was very specific on this; they don't want the metadata to be visible on GitHub.

This is still handled in the new tooling?

Comment (SvenDowideit) [inline touch-up.sh]: yup, hugo 0.15-dev has merged a PR that I made to allow commented out metadata :)

Comment (thaJeztah) [inline touch-up.sh]: Sweet!

File: Makefile [hunk 2]
```diff
@@ -47,3 +53,47 @@ export: build
...
+compose:
+	docker run --rm -it \
+		-e GITHUB_USERNAME -e GITHUB_TOKEN \
+		--env-file aws.env \
+		-v $(CURDIR):$(CURDIR) \
+		-v /var/run/docker.sock:/var/run/docker.sock \
+		-v /usr/bin/docker-static:/usr/bin/docker \
+		-w $(CURDIR) \
+		--entrypoint bash \
+			svendowideit/compose
```
Comment (dnephin) [inline Makefile]: We have an official image for this now: https://hub.docker.com/r/docker/compose/tags/

You probably want to use that?

Comment (dnephin) [inline Makefile]: Actually, I don't see this referenced anywhere, is it used?

Comment (thaJeztah) [inline Makefile]: Probably somewhere in Australia:

> Sven doesn't have docker-compose installed on some boxes, so use a compose container

**Comment by moxiegirl** [comment]
@SvenDowideit I think you want to merge this into hugo-github-linking. That is the branch we will build from as we have switched all the projects to your hugo linking. 

Also look at the PR @dnephin created. His fix stops the looping we were seeing.

File: Makefile [hunk 3]
```diff
@@ -47,3 +53,47 @@ export: build
...
+
+markdownlint:
+	docker exec -it docsdockercom_serve_1 /usr/local/bin/markdownlint /docs/content/
+
+htmllint:
+	docker exec -it docsdockercom_serve_1 /usr/local/bin/linkcheck http://127.0.0.1:8000
+
+all: clean build build-images serve
+
+# Sven doesn't have docker-compose installed on some boxes, so use a compose container
```
Comment (moxiegirl) [inline Makefile]: @SvenDowideit here is a personal looking bit in here.

File: Makefile (hunk 2 above)
Comment (SvenDowideit) [inline Makefile]: yeah - its a way to allow me to use compose based stuff - but massive \o/ for and official `docker/compose` image - I'll test that soon

**Comment by SvenDowideit** [comment]
roger, I'll close this and make a new PR to track from `hugo-github-linking` to master

**Pull request closed without merging**